api/tests/functional-tests/backend/query/test_ops.py

```diff
@@ -653,102 +676,157 @@ def model_sim(
-        Query("not_a_valid_attribute")
+    with pytest.raises(ArgumentError):
+        select_from_groundtruths("not a valid attribute")
 
     # Q: Get names for datasets where label class=cat exists in groundtruths.
-    f = schemas.Filter(labels=[{"class": "cat"}])
-    query_obj = Query(distinct(models.Dataset.name))
-    assert len(query_obj._selected) == 1
-    q = query_obj.filter(f).groundtruths()
-
-    dataset_names = db.query(q).all()  # type: ignore - SQLAlchemy type issue
+    f = Filter(groundtruths=create_label_filter(key="class", value="cat"))
+    dataset_names = select_from_groundtruths(
+        distinct(models.Dataset.name), filter_=f
+    )
+    dataset_names = db.query(dataset_names.subquery()).all()
     assert len(dataset_names) == 1
     assert (dset_name,) in dataset_names
 
     # Q: Get names for datasets where label=tree exists in groundtruths
-    f = schemas.Filter(labels=[{"class": "tree"}])
-    q = Query(models.Dataset.name).filter(f).groundtruths()
-    dataset_names = db.query(q).all()  # type: ignore - SQLAlchemy type issue
+    f = Filter(groundtruths=create_label_filter(key="class", value="tree"))
+    dataset_names = select_from_groundtruths(
+        distinct(models.Dataset.name), filter_=f
+    )
+    dataset_names = db.query(dataset_names.subquery()).all()
     assert len(dataset_names) == 0
 
 
 def test_query_models(
     db: Session,
     model_sim,
 ):
-    # Q: Get names for all models that operate over a dataset that meets the name equality
-    f = schemas.Filter(
-        dataset_names=[dset_name],
-    )
-    q = Query(models.Model.name).filter(f).any()
-    model_names = db.query(q).distinct().all()  # type: ignore - SQLAlchemy type issue
-    assert len(model_names) == 2
-    assert (model_name1,) in model_names
-    assert (model_name2,) in model_names
-
-    # Q: Get names for models where label=cat exists in predictions
-    f = schemas.Filter(labels=[{"class": "cat"}])
-    q = Query(models.Model.name).filter(f).predictions()
-    model_names = db.query(q).distinct().all()  # type: ignore - SQLAlchemy type issue
+    # Q: Get names for all models that operate over a dataset.
+    f = Filter(
+        predictions=create_dataset_filter(dset_name),
+    )
+    model_names = select_from_annotations(
+        models.Model.name, filter_=f
+    ).distinct()
+    model_names = db.query(model_names.subquery()).all()
     assert len(model_names) == 2
     assert (model_name1,) in model_names
     assert (model_name2,) in model_names
 
-    # Q: Get names for models where label=tree exists in predictions
-    f = schemas.Filter(labels=[{"class": "tree"}])
-    q = Query(models.Model.name).filter(f).predictions()
-    model_names = db.query(q).distinct().all()  # type: ignore - SQLAlchemy type issue
+    # Q: Get names for models that operate over dataset that doesn't exist.
+    f = Filter(predictions=create_dataset_filter("invalid"))
+    model_names = select_from_annotations(
```

Comment:
nit: it'd be easier to follow these tests if they were split up by groundtruths / prediction / etc. (i.e., `test_select_from_predictions`, `test_select_from_groundtruths`, etc.)

Reply:
the overall test coverage here seems solid, so I'm not too worried about this one

Reply:
Ill make a note of this for the integration tests in the next pr